src/test/kotlin/org/rust/lang/core/type/RsTypificationTestBase.kt

```diff
@@ -5,14 +5,31 @@
 
 package org.rust.lang.core.type
 
+import com.intellij.openapi.vfs.VirtualFileFilter
 import org.intellij.lang.annotations.Language
+import org.rust.fileTreeFromText
 import org.rust.lang.RsTestBase
 import org.rust.lang.core.psi.RsExpr
 import org.rust.lang.core.types.type
 
 abstract class RsTypificationTestBase : RsTestBase() {
     protected fun testExpr(@Language("Rust") code: String, description: String = "") {
         InlineFile(code)
+        check(description)
+    }
+
+    protected fun stubOnlyTypeInfer(@Language("Rust") code: String, description: String = "") {
+        val testProject = fileTreeFromText(code)
+            .createAndOpenFileWithCaretMarker()
+
+        checkAstNotLoaded(VirtualFileFilter { file ->
+            !file.path.endsWith(testProject.fileWithCaret)
```

Comment:
Have you verified that this assert really fires if type inferences touches AST?

Reply:
At least, my current implementation of const expr evaluation fails:)

Reply:
:)